java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/config/AzureServiceBusSourceConfig.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.config;
+
+import io.lenses.streamreactor.common.config.base.BaseConfig;
+import io.lenses.streamreactor.common.config.base.intf.ConnectorPrefixed;
+import java.util.Map;
+import lombok.Getter;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Width;
+
+/**
+ * Service Bus Connector Configuration class.
+ */
+public class AzureServiceBusSourceConfig extends BaseConfig implements ConnectorPrefixed {
+
+  @Getter
+  private static final ConfigDef configDefinition;
+
+  public static final String CONNECTION_GROUP = "Connection";
+  public static final String BASE_GROUP = "Base";
+
+  static {
+    configDefinition =
+        new ConfigDef()
+            .define(AzureServiceBusConfigConstants.CONNECTOR_NAME,
+                Type.STRING,
+                AzureServiceBusConfigConstants.CONNECTOR_NAME_DEFAULT,
+                Importance.HIGH,
+                AzureServiceBusConfigConstants.CONNECTOR_NAME_DOC,
+                BASE_GROUP,
+                1,
+                ConfigDef.Width.LONG,
+                AzureServiceBusConfigConstants.CONNECTOR_NAME
+            ).define(AzureServiceBusConfigConstants.TASK_RECORDS_QUEUE_SIZE,
+                Type.INT,
+                AzureServiceBusConfigConstants.TASK_RECORDS_QUEUE_SIZE_DEFAULT,
+                Importance.MEDIUM,
+                AzureServiceBusConfigConstants.TASK_RECORDS_QUEUE_SIZE_DOC,
+                BASE_GROUP,
+                2,
+                Width.SHORT,
+                AzureServiceBusConfigConstants.TASK_RECORDS_QUEUE_SIZE
+            ).define(AzureServiceBusConfigConstants.CONNECTION_STRING,
+                Type.STRING,
+                Importance.HIGH,
+                AzureServiceBusConfigConstants.CONNECTION_STRING_DOC,
+                CONNECTION_GROUP,
+                2,
+                ConfigDef.Width.LONG,
+                AzureServiceBusConfigConstants.CONNECTION_STRING
+            ).define(AzureServiceBusConfigConstants.KCQL_CONFIG,
```

Comment:
I wanted to point you to [this](https://github.com/lensesio/stream-reactor/blob/26d8f2652e8197a7aa1cc5cc7c4311194414969b/kafka-connect-azure-datalake/src/main/scala/io/lenses/streamreactor/connect/datalake/auth/DatalakeClientCreator.scala#L41)

Customers won't always be able to use connection string for connecting to azure, especially if they are running on an Azure instance in which they will have default chain from the instance profile.  I believe the original PR was more complete in this respect?

But if we can we should identify code we can reuse here and move it into the Java repo so we are not inventing the wheel and duplicating throughout the codebase.  I did similar recently for GCP cloud connectors, so can help with this if you would like.

Reply:
There are two ways described in official Service Bus docs: one being **Passwordless**, second being **Connection String**. Since the first one will work using specifically-installed Azure commandline and our connectors are being ran inside docker container, it would require us to include Azure commandline and ways to work with it within Lenses Box (which will no doubt be huge amounts of work).

Connection String already proved to work for our clients in other Azure connectors, so that's why I've chosen to implement this path.

My take on it is: unless we get a specific requirement to support another type of authentication, let's just use what we have (as this is another feature that we don't know if ever will be used yet)